internal/mesh/internal/controllers/sidecarproxy/builder/destinations.go

```diff
@@ -337,7 +337,7 @@ func (b *Builder) buildDestination(
 			}
 		}
 
-		b.addCluster(clusterName, endpointGroups, connectTimeout)
+		b.addCluster(clusterName, endpointGroups, connectTimeout, pbproxystate.Protocol(effectiveProtocol))
```

Comment:
the pbproxystate.protocol matches the pbcatalog enum so rather than maintaining a manual map, we can just cast it.  if the two enums start to change independently, the golden tests will catch will catch it and cause action to be taken to keep them in sync.

Reply:
SGTM. I went ahead and added a small drift detection test to be more explicit so folks don't just fix the goldens blindly.